client/src/request.js

```diff
@@ -1,38 +1,63 @@
-const photos = (productId) => {
-  return fetch(`http://ec2-3-136-203-39.us-east-2.compute.amazonaws.com:4002/photos/id/${productId}`)
-  .then(res => res.json())
-  .then(productPhotos => {
-    return productPhotos;
-  })
-  .catch(error => {
+const fetchWithTimeout = async (resource, options) => {
+  const { timeout = 4000 } = options;
+
+  const controller = new AbortController();
+  const id = setTimeout(() => controller.abort(), timeout);
+
+  const response = await fetch(resource, {
+    ...options,
```

Comment:
What is "...options" something to be here?

Reply:
So this is using the spread syntax to pass all the key value pairs of the options object to the fetch request. The fetch request syntax accepts options such as: Method: GET, POST, PUT, etc. This way the options passed into the fetchWithTimeout (line 1) will be passed into the fetch function that's actually making the network request.